test/cases/azure.sh

```diff
@@ -0,0 +1,232 @@
+#!/bin/bash
+set -euo pipefail
+
+source /etc/os-release
+
+# Colorful output.
+function greenprint {
+    echo -e "\033[1;32m${1}\033[0m"
+}
+
+#TODO: Remove this once there is rhel9 support for Azure image type
+if [[ $DISTRO_CODE == rhel_90 ]]; then
+    greenprint "Skipped"
+    exit 0
+fi
+
+# Provision the software under tet.
+/usr/libexec/osbuild-composer-test/provision.sh
+
+# We need azure-cli to talk to Azure.
+if ! hash az; then
+    # this installation method is taken from the official docs:
+    # https://docs.microsoft.com/cs-cz/cli/azure/install-azure-cli-linux?pivots=dnf
+    sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+    echo -e "[azure-cli]
+name=Azure CLI
+baseurl=https://packages.microsoft.com/yumrepos/azure-cli
+enabled=1
+gpgcheck=1
+gpgkey=https://packages.microsoft.com/keys/microsoft.asc" | sudo tee /etc/yum.repos.d/azure-cli.repo
+
+  greenprint "Installing azure-cli"
+  sudo dnf install -y azure-cli
+  az version
+fi
+
+# We need terraform to provision the vm in azure and then destroy it
+if [ "$ID" == "rhel" ] || [ "$ID" == "centos" ]
+then
+    release="RHEL"
+elif [ "$ID" == "fedora" ]
+then
+    release="fedora"
+else
+    echo "Test is not running on neither Fedora, RHEL or CentOS, terminating!"
+    exit 1
+fi
+sudo dnf install -y dnf-plugins-core
```

Comment:
I suggest making `dnf-plugins-core` a depedency of `osbuild-composer-tests` in the specfile.